plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/resource/ResourcePayloadPage.java

```diff
@@ -0,0 +1,204 @@
+                .applyTo(sourceGroup);
+        
+        createLocalSourceControls(dbc, sourceGroup);
+    }
+
+    private void createLocalSourceControls(DataBindingContext dbc, Group sourceGroup) {
+        Label label = new Label(sourceGroup, SWT.NONE);
+        GridDataFactory.fillDefaults().span(3, 1).applyTo(label);
+        label.setText("Enter a file path (workspace or local) or a full URL");
+        // local template file name
+        Text sourceText = new Text(sourceGroup, SWT.BORDER);
+        GridDataFactory.fillDefaults()
+                .align(SWT.FILL, SWT.CENTER).grab(true, false)
+                .applyTo(sourceText);
+        final IObservableValue source = WidgetProperties.text(SWT.Modify).observe(sourceText);
+        Binding binding = ValueBindingBuilder
+                .bind(source )
+                //.validatingBeforeSet(value->isFile(value.toString())?
+                //        ValidationStatus.ok(): 
+                //            ValidationStatus.error(value +" is not a file"))
+                .to(BeanProperties.value(
+                        IResourcePayloadPageModel.PROPERTY_SOURCE).observe(model))
+                .in(dbc);
+        MultiValidator validator = new MultiValidator() {
+            
+            @Override
+            protected IStatus validate() {
+                String sourceValue = (String) source.getValue();
+                return (boolean) !IResourcePayloadPageModel.URL_VALIDATOR.isValid(sourceValue) && !isFile(sourceValue)?ValidationStatus.error(sourceValue + " is not a file"):ValidationStatus.ok();
+            }
+        };
+        dbc.addValidationStatusProvider(validator);
+
+        // browse button
+        Button btnBrowseFiles = new Button(sourceGroup, SWT.NONE);
+        btnBrowseFiles.setText("File system...");
+        GridDataFactory.fillDefaults()
+                .align(SWT.LEFT, SWT.CENTER)
+                .applyTo(btnBrowseFiles);
+        UIUtils.setDefaultButtonWidth(btnBrowseFiles);
+
+        btnBrowseFiles.addSelectionListener(onFileSystemBrowseClicked());
+        
+        // browse button
+        Button btnBrowseWorkspaceFiles = new Button(sourceGroup, SWT.NONE);
+        btnBrowseWorkspaceFiles.setText("Workspace...");
+        GridDataFactory.fillDefaults()
+                .align(SWT.LEFT, SWT.CENTER)
+                .applyTo(btnBrowseWorkspaceFiles);
+        UIUtils.setDefaultButtonWidth(btnBrowseWorkspaceFiles);
+
+        btnBrowseWorkspaceFiles.addSelectionListener(onBrowseWorkspaceClicked());
+    }
+
+    private SelectionAdapter onFileSystemBrowseClicked() {
+        return new SelectionAdapter() {
+
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                FileDialog dialog = createFileDialog(model.getSource());
+                String file = dialog.open();
+                setLocalSourceFileName(file);
+            }
+
+            private FileDialog createFileDialog(String selectedFile) {
```

Comment:
May this dialog also limit resources with *.json as the other one that browses the workspace?